optimum/intel/openvino/modeling_text2speech.py

```diff
@@ -0,0 +1,678 @@
+            "speaker_embeddings": speaker_embeddings,
+            "encoder_hidden_states": encoder_last_hidden_state,
+            "encoder_attention_mask": encoder_attention_mask,
+            "beam_idx": np.arange(bsz, dtype=np.int32),
+        }
+        result = self.request(inputs)
+        output_sequence_out = torch.from_numpy(result[0])
+        spectrum = torch.from_numpy(result[1])
+        prob = torch.from_numpy(result[2])
+        return ModelOutput(output_sequence_out=output_sequence_out, spectrum=spectrum, prob=prob)
+
+
+class OVTextToSpeechPostNet(OVModelPart):
+    _model_name = "postnet_model"
+
+    def __init__(self, model: ov.Model, parent_model: OVBaseModel) -> None:
+        super().__init__(model, parent_model, model_name=self._model_name)
+        self.output_dtypes = {key.get_any_name(): key.get_element_type().get_type_name() for key in self.model.outputs}
+        self.output_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.outputs)}
+        self.input_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.inputs)}
+        self.hidden_states_output_names = []
+        if len(self.model.outputs) > 2:
+            self.hidden_states_output_names = [
+                key.get_any_name() for key in self.model.outputs[2:] if "hidden_states" in key.get_any_name()
+            ]
+
+    def forward(self, spectrograms, **kwargs):
+        self._compile()
+        inputs = {
+            "raw_spectrogram": spectrograms,
+        }
+        result = self.request(inputs)
+        postnet_spectrogram = torch.from_numpy(result[0])
+        return ModelOutput(postnet_spectrogram=postnet_spectrogram)
+
+
+class OVTextToSpeechVocoder(OVModelPart):
+    _model_name = "vocoder_model"
+
+    def __init__(self, model: ov.Model, parent_model: OVBaseModel) -> None:
+        super().__init__(model, parent_model, model_name=self._model_name)
+        self.output_dtypes = {key.get_any_name(): key.get_element_type().get_type_name() for key in self.model.outputs}
+        self.output_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.outputs)}
+        self.input_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.inputs)}
+        self.hidden_states_output_names = []
+        if len(self.model.outputs) > 2:
+            self.hidden_states_output_names = [
+                key.get_any_name() for key in self.model.outputs[2:] if "hidden_states" in key.get_any_name()
+            ]
+
+    def forward(self, spectrogram, **kwargs):
+        self._compile()
+        inputs = {
+            "spectrogram": spectrogram,
+        }
+        result = self.request(inputs)
+        waveform = torch.from_numpy(result[0])
+        return ModelOutput(waveform=waveform)
+
+
+@add_start_docstrings(
+    """
+    text-to-audio models
+    """,
+    INPUTS_DOCSTRING,
+)
+class OVModelForTextToSpeechSeq2Seq(OVModelForSeq2SeqLM):
+    auto_model_class = SpeechT5ForTextToSpeech
```

Comment:
this is specific to speecht5 so should be replaced, why not use `AutoModelForTextToSpectrogram` instead ?

Reply:
Maybe it will be better make similar like for SpeechToText (where general class serves as factory for creation model specific class for whisper
https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/modeling_seq2seq.py#L999

it will simplify enabling new models that required different behavior in future (as I understand, different text2speech may have different subcomponents and generation flow)

Reply:
Agreed with @eaidova. Created common interface for text-to-audio models using OpenVINO.
@echarlaix, changed to `AutoModelForTextToWaveform` becuase spectrogram looks raw (or unprocessed) format of audio